lib/groovehq/resource.rb

```diff
@@ -20,7 +20,7 @@ def initialize(client, data)
 
     def parse_links(links)
       (links || {}).each_with_object({}) do |(relation, value), result|
-        result[relation] = Relation.new(@client, value["href"])
+        result[relation] = Relation.new(@client, value["href"] || value)
```

Comment:
This change belongs to `.initialize`, where self relation is being created. This relation should initially be a regular relation with "href" attribute. Please update `initialize` to build `self` like this:

```ruby
      links            = data.delete(:links) { ActiveSupport::HashWithIndifferentAccess.new }
      links[:self]   = { href: data.delete(:href) } if data.has_key?(:href)
```

And then inside "parse_links" you can also change `value["href"]` to `value[:href]` so keys access is consistent across whole file.

Thanks! Good catch with this bug.